azurerm/resource_arm_virtual_network_gateway_connection.go

```diff
@@ -180,7 +289,9 @@ func resourceArmVirtualNetworkGatewayConnectionRead(d *schema.ResourceData, meta
 		d.Set("virtual_network_gateway_id", conn.VirtualNetworkGateway1.ID)
 	}
 
-	d.Set("authorization_key", conn.AuthorizationKey)
+	if conn.AuthorizationKey != nil {
+		d.Set("authorization_key", conn.AuthorizationKey)
```

Comment:
providing this value is a string, we shouldn't need this check?

Reply:
It is an optional field in the API, is it then also safe to avoid this check?

Reply:
Since `d.Set` handles setting pointer values (even if they're nil - to an empty value) - it should be fine :)